### PR TITLE
delete needs ids

### PR DESCRIPTION
**Pull request by KishaMims**
Line: 32 I have a new obj made that is called meals that is taking in my usermeals data from the fetch. This loop over helps remove any "duplicate" meal course names i..e Breakfast, Lunch etc. I am confused on how to add the id's in for each food entry, though, as I need them to know which id to send in the delete query in my db.  I thought there probably needs to be another nested obj inside for each food eaten, but I don't know how the logic would go for that.

### CODE REVIEW

**Comment by cdm-ium** [comment]
It would help if you could try breaking this down into incremental pieces. 

Do you have a REST API endpoint working for deletes? Maybe you could start with that and break it down to a single API request. That is/is not working. I am guessing you have done an example of a DELETE API request somewhere before and so you could try modifying that to work with your app.

Next you could add a component/button for the deletebutton. Maybe you have that on main already but I cannot see anything new besides your comment in the diff.

I will try taking another look this evening, but if you could show me a bit more of things you've tried it would help me know exactly where you're going wrong. I'm sure you're closer than you think already!

**Comment by KishaMims** [comment]
Okay, I don't think I explained what I was asking well. I have that new object I made meals on like 32 - this is what is displayed on my page. When making the object meal course = meal course, I wanted to deleted repeat titles. However, in doing this, I do not know how to add in the ID (I can only have one key if I put in the id, it will write over the foodeaten text) - I need the specific ID to delete the food since I have to tell the sql query where to delete from. Also, I realized if I have multiples of one food item, let's say eggs - it is posted on my db, but only one will show with how I have this new obj set up. I need to know if I can add in each foodeaten  id (React is upset because I am missing the li unique keys as well) and I guess maybe add a counter++ if the fooeaten is there multiple times?

File: client/src/pages/userview.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ const navigateToAddMeal = () => {
   navigate(`/api/nutrition`)
 }
 
-
+//unsure how to add id back in here need them for delete all need to to have as keys ??
 let meals = {};
```
Comment (cdm-ium) [inline client/src/pages/userview.js]: To make it a little easier to understand what you intend the structure of `meals` to be, you might want to do:

```
const meals = {
  Breakfast: [],
  Lunch: [],
  Dinner: [],
  Snacks: [],
};
```

File: client/src/pages/userview.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ const navigateToAddMeal = () => {
...
 let totalCalories = 0;
```
Comment (cdm-ium) [inline client/src/pages/userview.js]: There are a bunch of ways you could do it, but here's one idea that might help:

```
usermeals.forEach((meal) => {
  meals[entry.mealcourse].push(meal);
  totalCalories += Number(meal.calories)
});
```

Comment (cdm-ium) [reply client/src/pages/userview.js]: Then on line 51 you could do something like
```
 Object.entries(meals.Breakfast).map((food) => {
          return (<li id={food.id}>{food.foodeaten} CAL:{food.calories}</li>);
        })

```

I would recommend using objects over lists so it is easier to read the code. When I read you code and saw `food[0]` or `food[1]` I had to guess what those items are, plus in the original map you are doing above, you are throwing the `id` away which you want for the `li` elements.

**Comment by KishaMims** [comment]
Fixed data display and delete working and complete

**Pull request merged**